lib/mocha.js

```diff
@@ -471,7 +491,14 @@ Mocha.prototype.run = function(fn) {
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function(reporterConfig) {
+    var _reporter = reporterConfig.fn;
+    options.reporterOptions = reporterConfig.options;
```

Comment:
Any support for cloning the `options` object? Can't relay on `Object.assign`, no polyfill nor dependency that includes a similar utility. Should I go for a `JSON.stringify`/`parse` route?